src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(new Random(4000).nextLong());
+        } catch (InterruptedException e) {
+            System.out.println("Interrupted for " + userId);
+        }
+        UserData cashedUserData = cache.get(userId);
+        if (cashedUserData != null) {
+            return cashedUserData;
+        }
+        UserData userData = new UserData(userId, "Details for " + userId);
+        cache.put(userId, userData);
+        return userData;
```

Comment:
The sleep method is used to simulate a delay, but this will block the thread. In an asynchronous context, this is not ideal as it can lead to inefficient thread utilization. Consider using a scheduled executor to simulate the delay without blocking, or if the delay is meant to mimic network latency or a long-running computation, model it appropriately without using sleep.